docs/builtins.md

```diff
@@ -1742,3 +1742,287 @@ Output
+
+### `dash`
+
+`dash` sets the line dash pattern.
+
+#### Example
+
+```evy
+width 2
+
+dash 5 // same as: dash 5 5, dash 5 5 5
+hline 85 "red"
+
+dash 10 4 1 4
+hline 75 "blue"
+
+dash 10 5 10 // same as: dash 10 5 10 10 5 10
+hline 65 "gold"
+
+dash // reset dash
+hline 50 "black"
+
+gridn 5 "gray"
+
+func hline y:num c:string
+    color c
+    move 0 y
+    line 100 y
+end
+```
+
+Output
+
+<img width="300" alt="dash patterns" src="img/dashes.png">
+
+#### Reference
+
+    dash segments:num...
+
+The `dash` function sets the line dash pattern used when stroking lines.
+The dash pattern is specified as a variadic number of arguments, where
+each argument represents the length of a dash or gap. For example, the
+arguments `5 10` would create a line with 5-unit long dashes and 10-unit
+long gaps.
+
+If the number of arguments is odd, they are copied and concatenated. For
+example, the arguments `10 5 10` would become `10 5 10 10 5 10`. If no
+arguments are given, the line returns to being solid.
+
+The initial dash pattern is a solid line.
+
+### `linecap`
+
+`linecap` sets the shape of the ends of lines.
+
+#### Example
+
+```evy
+width 5
+grid
+
+linecap "round"
+hline 70
+
+linecap "butt"
+hline 50
+
+linecap "square"
+hline 30
+
+func hline y:num
+    move 10 y
+    line 90 y
+end
+```
+
+Output
+
+<img width="300" alt="round and square line caps" src="img/linecaps.png">
+
+#### Reference
+
+    linecap style:string
+
+The `linecap` sets the shape of the ends of lines to the `style` string
+argument. Valid styles are `"round"`, `"butt"` or `"square"`. An
+invalid style takes no effect.
+
+Here is a table of the different linecap shapes and their descriptions:
+
+Shape Description
+
+| Shape      | Description                                                                                                           |
+| ---------- | --------------------------------------------------------------------------------------------------------------------- |
+| `"round"`  | The ends of the line are rounded.                                                                                     |
+| `"butt"`   | The ends of the line are squared off at the endpoints.                                                                |
+| `"square"` | The ends of the line are squared off by adding a box with an equal width and half the height of the line's thickness. |
+
+The initial linecap style is `"round"`.
+
+### `text`
+
+`text` prints text to the canvas at the current cursor position.
+
+#### Example
+
+```evy
+move 20 70
+text "“Time is an illusion."
+move 20 63
+text "Lunchtime doubly so.”"
+move 35 48
+text "― Douglas Adams"
+```
+
+Output
+
+<img width="300" alt="text sample" src="img/text.png">
+
+#### Reference
+
+    text s:string
+
+The `text` function prints the string argument `s` to the canvas at the
+current cursor position. The cursor position is not updated after
+writing text. Only the `fill` and `color` functions update the text
```

Comment:
Perhaps instead of "update" in this sentence, you say: "Only `fill` and `color` have an effect on the text; `stroke` has no effect."

Reply:
done.